### ISSUE TITLE
Mail alias generator

### ISSUE THREAD

**Issue by Safranil** [mail-in-a-box/mailinabox#2094]
Hi,

Is it possible to implement a "Firefox Relay" like functionality in Mail In A Box and allow users of the box to get anonymized/trash mail address?

**Comment by kurucu** [comment]
This is just creating an alias for yourself right, so you could do it through the admin panel easily? If you wanted to send from it too, then you just need to add the alias as a sender ID to roundcube or to your email client.
Beyond that, for a more integrated experience, MAIB includes all the APIs necessary to do this - you just need to build an app/interface/extension around them!

**Comment by kurucu** [comment]
It's actually quite a nice idea to build a browser extension that takes your mail credentials and lets you generate relay (alias) addresses on the fly - but I wouldn't know where to start.

**Comment by nomandera** [comment]
Using the `mailbox`+`random@example.com` is not especially good match for this request but it has the advantage that it works as is.

I tend to use `mailbox`+`shopOrSiteName@example.com` for everything

**Comment by Safranil** [comment]
Yeah but some website refuse the email if it contains a + in it.

The extension is also a good idea.

Le lun. 21 févr. 2022 à 15:18, nomandera ***@***.***> a
écrit :

**Comment by Synchro** [comment]
Oh, I know that - I'm the maintainer of [this](https://youdontknowemail.tumblr.com)! (Though I've not bothered updating it for a long time)

**Comment by nomandera** [comment]
> 

Yes this is very much true. I estimate about 2-3% of my attempts to use have been rejected due to bad mail validation checking.

I only suggest it as

**Comment by kurucu** [comment]
> Using the `mailbox`+`random@example.com` is not especially good match for this request but it has the advantage that it works as is.
> 
> I tend to use `mailbox`+`shopOrSiteName@example.com` for everything

This would be my go-to, but I also find very few sites actually allow + to pass validation.

ALSO of course, you can't just delete the address to shut them off when they leak your address to spambots, whereas with an alias you can be as agile as you like.

**Comment by kurucu** [comment]
I've dropped a discussion [here](https://discourse.mailinabox.email/t/private-email-hide-my-email-browser-extension/8833), trying to canvas interest in such a thing!

**Comment by tdulcet** [comment]
I am a Firefox [add-on developer](https://addons.mozilla.org/firefox/user/10108346/). This is a great idea and the extension would be easy to implement if there was sufficient interest. However, I think the Mail in a Box API would first need to be updated to no longer require an admin account, so that more people would be able to use it.
